### DISCUSSION TITLE
Analysis Data I/O

### DISCUSSION THREAD

**Discussion by samussiah** [Gilead-Public/gsm#1792] [Ideas]
Where does historical analysis data fit into the {gsm} analysis workflow?

TL;DR: each workflow ends with an implicit data store, including data mapping, data aggregation, and data analysis.  Each data layer is stored in a hierarchy by study and snapshot date.

{gsm} generates a single cross-sectional "snapshot" of analysis data by running a set of metric workflows.  This analysis data drives the data visualizations and reports, along with metric and group metadata.

In between generating the analysis data and running the visualizations and reports the new snapshot needs to be stored and the previous snapshots need to be pulled in.  {gsm} knows what to do with longitudinal analysis data but it probably shouldn't be concerned with saving and reading it.

**Comment by samussiah** [comment]
Leaning toward an implicit data store at the end of each workflow by default, with support for an explicit specification of which data objects are stored.
